resources/lib/xmlfunctions.py

```diff
@@ -717,13 +717,61 @@ def buildElement( self, item, Tree, groupName, visibilityCondition, profileVisib
                         additionalproperty.text = DATA.local( self.MAINBACKGROUND[ key ].decode( "utf-8" ) )[1]
                     except:
                         additionalproperty.text = DATA.local( self.MAINBACKGROUND[ key ] )[1]
+
+        propertyPatterns = self.getPropertyPatterns(labelID.text, groupName)
+        if len(propertyPatterns) > 0:
+            propertyReplacements = self.getPropertyReplacements(newelement)
+            for propertyName in propertyPatterns:
+                propertyPattern = propertyPatterns[propertyName][0]
+                for original, replacement in propertyReplacements:
+                    regexpPattern = re.compile(re.escape(original), re.IGNORECASE)
+                    propertyPattern = regexpPattern.sub(replacement, propertyPattern)
+    
+                additionalproperty = xmltree.SubElement(newelement, "property")
+                additionalproperty.set("name", propertyName.decode("utf-8"))
+                additionalproperty.text = propertyPattern.decode("utf-8")
             
         return newelement
+    
+    def getPropertyPatterns(self, labelID, group):
+        overrides = DATA._get_overrides_skin()
+        propertyPatterns = {}
+        
+        if overrides is not None:
+            propertyPatternElements = overrides.getroot().findall("propertypattern")
+            for propertyPatternElement in propertyPatternElements:
+                propertyName = propertyPatternElement.get("property")
+                propertyGroup = propertyPatternElement.get("group")
+              
+                if propertyName and propertyGroup and propertyPatternElement.text:
+                    if propertyGroup == group:
```

Comment:
would be nice you would do if not ...: continue instead of super-deep nesting :)